python/pyiceberg/io/pyarrow.py

```diff
@@ -366,7 +352,7 @@ def visit_string(self, _: StringType) -> pa.DataType:
         return pa.string()
 
     def visit_uuid(self, _: UUIDType) -> pa.DataType:
-        return UuidType()
+        return pa.binary(16)
```

Comment:
this makes sense, we're doing the same in https://github.com/apache/iceberg/blob/c07f2aabc0a1d02f068ecf1514d2479c0fbdd3b0/arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java#L325-L327